### PR TITLE
Remove some unused includes in code maintained by NXP

### PR DESCRIPTION
**Pull request by clementfaure**

### CODE REVIEW

File: core/drivers/crypto/crypto_api/acipher/rsassa.c [hunk 1]
```diff
@@ -247,10 +247,15 @@ static TEE_Result rsassa_pkcs1_v1_5_verify(struct drvcrypt_rsa_ssa *ssa_data)
...
+	if (!EM.data) {
+		ret = TEE_ERROR_OUT_OF_MEMORY;
+		goto end_verify;
+	}
+
 	EM_gen.length = ssa_data->key.n_size;
 	EM_gen.data = malloc(EM.length);
 
-	if (!EM.data || !EM_gen.data) {
+	if (!EM_gen.data) {
```
Comment (etienne-lms) [inline core/drivers/crypto/crypto_api/acipher/rsassa.c:258]: I don't understand what issue is addressed here.

**Review by etienne-lms — COMMENTED** [review]
`Reviewed-by: Etienne Carriere <etienne.carriere@linaro.org>` for commit
"core: crypto_api: remove unused".

`Acked-by: Etienne Carriere <etienne.carriere@linaro.org>` for commits
"core: imx: remove unused includes" and
"drivers: caam: remove unused includes".

File: core/drivers/crypto/crypto_api/acipher/rsassa.c (hunk 1 above)
Comment (clementfaure) [reply core/drivers/crypto/crypto_api/acipher/rsassa.c:258]: Basically, Coverity was not happy about having `EM.data` not checked right after `malloc()` call. It reports a `CERT-C ERR33-C` warning. I believe I also got this warning from `clang-tidy`

Comment (etienne-lms) [reply core/drivers/crypto/crypto_api/acipher/rsassa.c:258]: Thank the It looks like a false positive here. Maybe put this description in the commit log, and whether it is a
CERTC-C error or warning.
I believe there are other occurrences in existing optee_os implementation. Do you plan to change them all?

Comment (clementfaure) [reply core/drivers/crypto/crypto_api/acipher/rsassa.c:258]: Ok, I did explain it in the commit message.
According to Coverity it's a coding violation.

Well, I only allow myself to fix Coverity issues on code maintained by NXP. At NXP, we have custom rules and requirements regarding Coverity. What is a Coverity/coding issue for us, might be anecdotal for you, optee-os maintainers.
Right now, our scans report about 400 issues on optee-os (crypto libs excluded). Most of them have negligible impact/false reports.
I fix NXP code, but I also keep an eye on the rest of the code base. (like d3ed98ee7).
So yes, I would be happy to help but if you guys could share what your requirements are that would be great.

Comment (etienne-lms) [reply core/drivers/crypto/crypto_api/acipher/rsassa.c:258]: Ok. As far as I'm concerned: `Acked-by: Etienne Carriere <etienne.carriere@linaro.org>` for that commit "core: crypto_api: fix malloc() allocation check". @jforissier, @jenswi-linaro, your thoughts?

Comment (jforissier) [reply core/drivers/crypto/crypto_api/acipher/rsassa.c:258]: As far as I'm concerned, I'm OK with that bit. It doesn't make much sense to call the second `malloc()` if the first failed anyways.

Now more generally speaking of Coverity: we've run it a couple of years ago and there were too many false positives to get any meaningful output. But it certainly can detect real issues so it's fine to address them in PRs as they come up.

Comment (jenswi-linaro) [reply core/drivers/crypto/crypto_api/acipher/rsassa.c:258]: Yes, there were a few false positives with Coverity but it was still possible to find a few issues to address.

**Review by jenswi-linaro — COMMENTED** [review]
`Acked-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Pull request merged**